File: csharp/src/Core/Helpers/RedisHelper.cs
namespace Train.Solver.Core.Helpers;

public static class RedisHelper
{
    public const string Delimiter = ":";

    public static string BuildLockKey(string network, string address, string? asset = null) 
        => $"{network}{Delimiter}{address}{(asset is not null ? $"{Delimiter}{asset}" : "")}";

    public static string BuildNonceKey(string network, string address, string? asset = null) 
        => $"{network}{Delimiter}{address}{Delimiter}{(asset is not null ? $"{asset}{Delimiter}" : "")}currentNonce";
}

File: js/src/CoreAbstraction/Infrastructure/RedisHelper/RedisHelper.ts
export const Delimiter = ":";

export function BuildLockKey(network: string, address: string, asset?: string): string {
    return asset
        ? `${network}${Delimiter}${address}${Delimiter}${asset}`
        : `${network}${Delimiter}${address}`;
}

export function BuildNonceKey(network: string, address: string, asset?: string): string {
    return asset
        ? `${network}${Delimiter}${address}${Delimiter}${asset}${Delimiter}currentNonce`
        : `${network}${Delimiter}${address}${Delimiter}currentNonce`;
}